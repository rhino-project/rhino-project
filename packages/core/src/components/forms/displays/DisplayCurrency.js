import { useGlobalComponent } from '../../../hooks/overrides';
import { InputGroup } from 'reactstrap';
import { applyCurrencyMask } from '../../../utils/ui';
import DisplayString from './DisplayString';

export const DisplayCurrencyBaseInput = (props) => {
  return (
    <DisplayString
      type="text"
      accessor={applyCurrencyMask}
      readOnly
      {...props}
    />
  );
};

export const DisplayCurrencyBase = (props) => {
  return (
    <InputGroup>
      <span className="input-group-text">$</span>
      <DisplayCurrencyBaseInput {...props} />
    </InputGroup>
  );
};

export const DisplayCurrency = (props) =>
  useGlobalComponent('DisplayCurrency', DisplayCurrencyBase, props);
