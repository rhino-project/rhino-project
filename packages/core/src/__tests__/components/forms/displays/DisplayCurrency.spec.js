import { sharedDisplayTests } from './sharedDisplayTests';
import DisplayCurrency from 'rhino/components/forms/displays/DisplayCurrency';

describe('DisplayCurrency', () => {
  sharedDisplayTests(DisplayCurrency);
});
