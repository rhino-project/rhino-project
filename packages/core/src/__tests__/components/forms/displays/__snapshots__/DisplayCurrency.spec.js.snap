// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisplayCurrency > renders inside of model context 1`] = `
<DocumentFragment>
  <div
    class="input-group"
  >
    <span
      class="input-group-text"
    >
      $
    </span>
    <input
      aria-invalid="false"
      autocomplete="off"
      class="form-control"
      id="dummy"
      name="dummy"
      path="dummy"
      readonly=""
      type="text"
      value="-"
    />
  </div>
</DocumentFragment>
`;

exports[`DisplayCurrency > should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  <div
    class="input-group"
  >
    <span
      class="input-group-text"
    >
      $
    </span>
    <input
      aria-invalid="false"
      autocomplete="off"
      class="form-control"
      id="dummy"
      name="dummy"
      path="dummy"
      readonly=""
      type="text"
      value="-"
    />
  </div>
</DocumentFragment>
`;

exports[`DisplayCurrency > should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  <div
    class="input-group"
  >
    <span
      class="input-group-text"
    >
      $
    </span>
    <input
      aria-invalid="false"
      autocomplete="off"
      class="form-control"
      id="dummy"
      name="dummy"
      path="dummy"
      readonly=""
      type="text"
      value="baz"
    />
  </div>
</DocumentFragment>
`;

exports[`DisplayCurrency > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
